mne/utils.py

```diff
@@ -381,7 +382,13 @@ def estimate_rank(data, tol=1e-4, return_singular=False,
         norms = _compute_row_norms(data)
         data /= norms[:, np.newaxis]
     s = linalg.svd(data, compute_uv=False, overwrite_a=True)
-    rank = np.sum(s >= tol)
+    if isinstance(tol, string_types):
+        if tol != 'auto':
+            raise ValueError('tol must be "auto" or float')
+        eps = np.finfo(float).eps
+        tol = np.max(data.shape) * np.amax(s) * eps
```

Comment:
This is the more principled fix. Turns out the bug was really with our `rank` calculations. Although our rank calculations had been wrong, when they were used later, we got lucky that the eigenvalues were small *and* negative, e.g. `-1e-26` for MEG covariance (where non-zero ones live around 1e-12), and our code used the check `nzero = (eig > 0)` in compute_whitener. But for whatever reason, on this 2.7 build they were small but positive, e.g. `1e-26` so the check failed to exclude them. Fixing the rank calculation this way -- by setting the tolerance to a reasonable value in the same way as `linalg.orth` does -- fixes the rank estimation, which then sets the `eig` value to zero explicitly in `_get_ch_whitener`. Whew!